plugin/pkg/auth/authenticator/token/oidc/oidc.go

```diff
@@ -238,7 +238,21 @@ func (a *OIDCAuthenticator) AuthenticateToken(value string) (user.Info, bool, er
 	var username string
 	switch a.usernameClaim {
 	case "email":
-		// TODO(yifan): Check 'email_verified' to make sure the email is valid.
+		// Only act on the "email_verified" claim if it's present. The lack of "email_verified"
```

Comment:
Since we haven't enforced this previously.

Most of the OIDC providers I know (google, salesforce, auth0, dex) support this claim, but I don't think UAA does[0]. The "standard claims" section is netoriously non-standard, and this seemed like a good way to not break users.

> the spec suggests that if this isn't explicitly true, it is false.

On second though I think that this might be a better option. We just may stop working with some providers.

[0] https://github.com/cloudfoundry/uaa/blob/2a4d48fe0b34f59c07d5ba176c945b52a9d7c65f/model/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/Claims.java